### PR TITLE
LPS-121658 Navigate to view the page /edit the content when clicking on the title of a content dashboard title

### PR DESCRIPTION
**Pull request by cgoncas**
### **Motivation**

After the user's test carried out in the LPS-117343 epic, we've confirmed that users expect to visit the page of the content when clicking on the title. Additionally, for those contents that have not an associated display page, we will do the title lead to the web content editor.

### **Proposed solution**

- When the web content has an associated display page and it has an approved status, even if the user has permission to edit it, the title will lead him to the page view, unless it has also a draft version of which the user is its owner (see next)
- When the web content has a draft version and the user is its owner, the title will lead him to the web content editor regardless of whether it has an associated display page or not.
- When the web content has not an associated display page and the user has permission to edit it, the title will lead him to the web content editor

### **How to test it**

Unit tests have been provided within this pull.

if you want to test it manually

_Appoved - display page web content will take to the view action_

1. Create a "web content" with the Admin user, associate a display page to it and publish it.
2. Go to Content Dashboard
3. Click on the title.
4. The link will take to the view action.

_Appoved - display page web content will take to the edit action_

1. Create a "web content" with the Admin user and publish it (without associate it to a display page).
2. Go to Content Dashboard
3. Click on the title.
4. The link will take to the edit action.

_Appoved/Draft - display page web content with owner user will take to the edit action_

1. Create a "web content" with the Admin user, associate a display page to it and publish it.
2. Edit the web content again and "Save as Draft".
3. Go to Content Dashboard
4. Click on the title.
5. The link will take to the edit action.

_Appoved/Draft - display page web content with owner user will take to the edit action_

1. Create a "web content" with the Admin user, associate a display page to it and publish it.
2. Edit the web content again and "Save as Draft".
3. log in as a User Admin different than the one that creates the Draft version of the web content
4. Go to Content Dashboard
5. Click on the title.
6. The link will take to the view action.

### CODE REVIEW

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
Hey @dgarciasarai !!

I need your help with this task.

Could you help me to use the right HTML tags and css styles to make the links looks like the ones in the FIGMA.

If you have any doubts about what I am talking about, please, let me know, so I can point to the right Mockup.

Thank you!!!

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
> Hey @dgarciasarai !!
> 
> I need your help with this task.
> 
> Could you help me to use the right HTML tags and css styles to make the links looks like the ones in the FIGMA.
> 
> If you have any doubts about what I am talking about, please, let me know, so I can point to the right Mockup.
> 
> Thank you!!!

Sure @cgoncas! I can help with this! 😄  I will look at the Figma styles and try to add it to your solution. In case I have any questions, I will ask you. Thanks!

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Review by dgarciasarai — APPROVED** [review]
Hey @cgoncas, I updated the JSP code to have the same styles as the design and I amend it to the last commit. Also, I tried manually all the cases and **there is one that is not working** for me: 

> _Appoved - display page web content will take to the edit action_
> 
> - Create a "web content" with the Admin user and publish it (without associate it to a display page).
> - Go to Content Dashboard
> - Click on the title.
> - The link will take to the edit action.

The others work perfectly 🎉 

Screenshot 2020-11-17 at 20 10 16

I suppose now it's time for a backend review :)

Thanks!

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
Hey @darquesdev !!

Would be awesome if you could take a look at this pull!!

Thanks!!!

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/JournalArticleContentDashboardItem.java [hunk 1]
```diff
@@ -188,6 +192,71 @@ public Date getCreateDate() {
 		).build();
 	}
 
+	@Override
+	public ContentDashboardItemAction getDefaultContentDashboardItemAction(
+		HttpServletRequest httpServletRequest) {
+
+		long userId = _portal.getUserId(httpServletRequest);
```
Comment (darquesdev) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/JournalArticleContentDashboardItem.java:199]: I'm not sure about this order, why did you use userId, locale and version?

File: modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/ViewJournalArticleContentDashboardItemAction.java [hunk 2]
```diff
@@ -98,6 +102,18 @@ private String _getViewURL(Locale locale, ThemeDisplay themeDisplay) {
 					JournalArticle.class.getName(),
 					_journalArticle.getResourcePrimKey(), locale,
 					clonedThemeDisplay)
+			).map(
+				url -> {
```
Comment (darquesdev) [inline modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/ViewJournalArticleContentDashboardItemAction.java:106]: i love it

File: modules/apps/content-dashboard/content-dashboard-web/src/test/java/com/liferay/content/dashboard/web/internal/item/JournalArticleContentDashboardItemTest.java [hunk 3]
```diff
@@ -160,6 +176,181 @@ public void testGetAssetTags() {
...
+						"http://localhost:8080/edit")),
+				null, null, null, journalArticle1, new LanguageImpl(),
+				journalArticle2, new PortalImpl());
+
+		ContentDashboardItemAction contentDashboardItemAction =
+			journalArticleContentDashboardItem.
+				getDefaultContentDashboardItemAction(
+					_getHttpServletRequest(RandomTestUtil.randomLong()));
+
+		Assert.assertEquals("viewURL", contentDashboardItemAction.getURL());
```
Comment (darquesdev) [inline modules/apps/content-dashboard/content-dashboard-web/src/test/java/com/liferay/content/dashboard/web/internal/item/JournalArticleContentDashboardItemTest.java:252]: Should it be ""http://localhost:8080/view""?

File: modules/apps/content-dashboard/content-dashboard-web/src/test/java/com/liferay/content/dashboard/web/internal/item/JournalArticleContentDashboardItemTest.java [hunk 4]
```diff
@@ -160,6 +176,181 @@ public void testGetAssetTags() {
...
+						"http://localhost:8080/edit")),
+				null, null, null, journalArticle1, new LanguageImpl(),
+				journalArticle2, new PortalImpl());
+
+		ContentDashboardItemAction contentDashboardItemAction =
+			journalArticleContentDashboardItem.
+				getDefaultContentDashboardItemAction(
+					_getHttpServletRequest(journalArticle2.getUserId()));
+
+		Assert.assertEquals("editURL", contentDashboardItemAction.getURL());
```
Comment (darquesdev) [inline modules/apps/content-dashboard/content-dashboard-web/src/test/java/com/liferay/content/dashboard/web/internal/item/JournalArticleContentDashboardItemTest.java:311]: Should it be "http://localhost:8080/edit"?

**Review by darquesdev — APPROVED** [review]
Great PR @cgoncas !! 
I just take a look at the code my portal was broken. I left a couple of comments/questions

File: modules/apps/content-dashboard/content-dashboard-web/src/test/java/com/liferay/content/dashboard/web/internal/item/JournalArticleContentDashboardItemTest.java (hunk 3 above)
Comment (cgoncas) [reply modules/apps/content-dashboard/content-dashboard-web/src/test/java/com/liferay/content/dashboard/web/internal/item/JournalArticleContentDashboardItemTest.java:252]: Yeahhh!!!!!

File: modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/ViewJournalArticleContentDashboardItemAction.java (hunk 2 above)
Comment (cgoncas) [reply modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/ViewJournalArticleContentDashboardItemAction.java:106]: :)

File: modules/apps/content-dashboard/content-dashboard-web/src/test/java/com/liferay/content/dashboard/web/internal/item/JournalArticleContentDashboardItemTest.java (hunk 4 above)
Comment (cgoncas) [reply modules/apps/content-dashboard/content-dashboard-web/src/test/java/com/liferay/content/dashboard/web/internal/item/JournalArticleContentDashboardItemTest.java:311]: Bad last minuute replacement 🤦

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
Thanks a lot for the review @darquesdev!!!

I have fixed the problems you found in the tests, so I am going to forward this PR.

Thanks!!!

**Pull request closed without merging**